modules/database/database_addons.bal

```diff
@@ -15,34 +12,54 @@ import ballerina/io;
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import ballerina/sql;
+import ballerina/io;
+
 public type IdSQL record {|
     int id;
 |};
 
+# Get experiment info
+#
+# + experimentId - experiment id
+# + return - experiment info
+public isolated transactional function getExperimentInfo(int experimentId) returns Experiment|error {
+    Experiment experimentInfo;
+    stream<record {|string name; string description;|}, sql:Error?> experimentResult = experimentDB->query(`SELECT name, description FROM Experiment WHERE experimentId=${experimentId};`);
+    var experiment = experimentResult.next();
+    check experimentResult.close();
+
+    if experiment is record {record {string name; string description;} value;} {
+        return {name: experiment.value.name, description: experiment.value.description};
+    } else {
+        return error(string `[experimentId ${experimentId}] Query to Experiment table unsuccessful.`);
+    }
+}
+
 # Clones an experiment in the database.
 #
 # + oldExperimentId - experiment id of the experiment that is to be cloned
-# + experiment - The data for the new experiment
 # + return - The experiment data including the database id or the encountered error
-public isolated transactional function cloneExperiment(int oldExperimentId, *Experiment experiment) returns ExperimentFull|error {
+public isolated transactional function cloneExperiment(int oldExperimentId) returns ExperimentFull|error {
     ExperimentFull? result = ();
 
-    // create new experiment (clone)
+    // clone experiment
+    Experiment experimentInfo = check getExperimentInfo(oldExperimentId);
     var experimentInsertResult = check experimentDB->execute(
-        `INSERT INTO Experiment (name, description) VALUES (${experiment.name}, ${experiment.description});`
+        `INSERT INTO Experiment (name, description) VALUES (${experimentInfo.name}, ${experimentInfo.description});`
```

Comment:
the cloned experiment should have an indicator like "(copy)" at the end of the name